### PR TITLE
Create an Auction Bidding on Avalanche using ReactJS

### PR DESCRIPTION
**Pull request by namannimmo10**
# Description

In this tutorial, we will be building a dApp through which we can organize auctions, bid in them and declare results, with both frontend and smart contracts. The interesting part about this tutorial is that we have used the Drizzle library for integrating the front end with blockchain. Drizzle is an awesome library developed by Trufflesuite, for seamless integration and it helps in keeping the data from blockchain updated in real-time.

# Boilerplate

Thanks for submitting this tutorial for Figment Learn!

Before you hit "Create Pull Request" make sure you have read the [Tutorial Contribution Guidelines](https://app.gitbook.com/@figment-learn/s/learn-docs/other/tutorial-guidelines). If you don't follow those guidelines it might take longer for us to review.

- [x] My tutorial follows the [Tutorial Contribution Guidelines](https://app.gitbook.com/@figment-learn/s/learn-docs/other/tutorial-guidelines).

In particular:

- [x] I've followed the [template structure](Structure).
- [x] I've proofread the tutorial for spelling and grammar mistakes.
- [x] I've ran a linter & prettifier to format my code blocks.
- [x] I've thoroughly explained the code, with inline code comments and text explanations.
- [x] I've ran the tutorial and the code works.

CC @rajranjan0608

### CODE REVIEW

**Comment by namannimmo10** [comment]
This is ready for review!  Thanks for your suggestions, @pyr0gan.

**Comment by pyr0gan** [comment]
> This is ready for review! Thanks for your suggestions, @pyr0gan.

Thanks @namannimmo10 I will review in the morning, when my eyes are fresh! 👀

**Comment by namannimmo10** [comment]
Looks like SUMMARY.md was deleted in upstream/master. Any reason for that?

**Comment by pyr0gan** [comment]
> Looks like SUMMARY.md was deleted in upstream/master. Any reason for that?

It's actually not needed anymore since we decoupled the Learn site from GitBook! 😄

**Review by pyr0gan — CHANGES_REQUESTED** [review]

File: network-documentation/avalanche/tutorials/create-an-auction-bidding-on-avalanche-using-reactjs.md [hunk 1]
```diff
@@ -0,0 +1,1085 @@
...
+const DATAHUB_RPC_URL = `https://avalanche--fuji--rpc.datahub.figment.io/apikey/${APIKEY}/ext/bc/C/rpc/`;
+
+module.exports = {
+  contracts_build_directory: "./client/src/build/contracts/",
+  networks: {
+    fuji: {
+      provider: function () {
+        return new HDWalletProvider({
+          mnemonic,
+          providerOrUrl: RPCURL,
```
[the remark this answers is not in the document]
Comment (namannimmo10) [reply network-documentation/avalanche/tutorials/create-an-auction-bidding-on-avalanche-using-reactjs.md:130]: There's already a const `DATAHUB_RPC_URL` for https://avalanche--fuji--rpc.datahub.figment.io/apikey/${APIKEY}/ext/bc/C/rpc/. Renaming it to `RPC_URL`.

[the remark this answers is not in the document]
Comment (namannimmo10) [reply network-documentation/avalanche/tutorials/create-an-auction-bidding-on-avalanche-using-reactjs.md:130]: > There's already a const `DATAHUB_RPC_URL`

Realized it was redundant...

> You can change this to `DATAHUB_RPC_URL` and remove `RPCURL` entirely

Done!

File: network-documentation/avalanche/tutorials/create-an-auction-bidding-on-avalanche-using-reactjs.md [hunk 2]
```diff
@@ -0,0 +1,1082 @@
...
+require("dotenv").config();
+const HDWalletProvider = require("@truffle/hdwallet-provider");
+
+// Account credentials from which our contract will be deployed
+const mnemonic = process.env.MNEMONIC;
+
+// API key of your Datahub account for Avalanche Fuji test network
+const APIKEY = process.env.APIKEY;
+
+const DATAHUB_RPC_URL = `https://api.avax-test.network/ext/bc/C/rpc`;
```
Comment (pyr0gan) [inline network-documentation/avalanche/tutorials/create-an-auction-bidding-on-avalanche-using-reactjs.md:120]: This is incorrect. I meant for you to keep the value of the DATAHUB_RPC_URL, not put the public RPC under this variable. Apologies if this was unclear from my comment.

It should be 
```text
DATAHUB_RPC_URL = `https://avalanche--fuji--rpc.datahub.figment.io/apikey/${process.env.APIKEY}/ext/bc/C/rpc`;
```

**Review by pyr0gan — APPROVED** [review]
Just one last change and it will be good to merge! Thank you for making the fixes 🚀

File: network-documentation/avalanche/tutorials/create-an-auction-bidding-on-avalanche-using-reactjs.md (hunk 2 above)
Comment (namannimmo10) [inline network-documentation/avalanche/tutorials/create-an-auction-bidding-on-avalanche-using-reactjs.md:120]:
```suggestion
const DATAHUB_RPC_URL = `https://avalanche--fuji--rpc.datahub.figment.io/apikey/${process.env.APIKEY}/ext/bc/C/rpc`;
```

**Comment by namannimmo10** [comment]
Thanks for the detailed review! I guess this can go in now.

**Pull request merged**